gtsam/geometry/Rot3.cpp

```diff
@@ -188,14 +188,18 @@ Vector Rot3::quaternion() const {
 
 /* ************************************************************************* */
 pair<Unit3, double> Rot3::axisAngle() {
-  const Vector3 omega = Rot3::Logmap(*this);
-  int direction = 1;
-  // Check if any element in axis is negative.
-  // This implies that the rotation is clockwise and not counterclockwise.
-  if (omega.minCoeff() < 0.0) {
-    direction = -1;
+  Vector3 omega = Rot3::Logmap(*this);
+  // Get the rotation angle.
+  double theta = omega.norm();
+
+  // Check if angle `theta` belongs to (-pi, pi].
+  // If yes, rotate in opposite direction to maintain range.
+  // Since omega = theta * u, if all coefficients are negative,
+  // then theta is outside the expected range.
+  if ((omega.array() < 0).all()) {
```

Comment:
Can you give some more explanation as to *why* this is the right way to normalize the angle? And why does just taking -theta work>? Because Unit3 constructor will normalize omega?